### PR TITLE
20170406 iperf3 metric fixes

### PR DESCRIPTION
**Pull request by grahamwhaley**
Combine two almost identical function/tests.
Add brief description comments

### CODE REVIEW

**Comment by grahamwhaley** [comment]
@dborquez Again, minor changes in the test output strings you may need to take into account in the metrics gathering parser.

File: tests/metrics/network/network-metrics-iperf3.sh [hunk 1]
```diff
@@ -58,24 +62,17 @@ function iperf3_jitter {
...
+function iperf3_bidirectional_bandwidth_client_server {
 	setup
 	bidirectional_bandwidth_result=$(mktemp)
 	$DOCKER_EXE run -d --name=iperf3-server ${image} bash -c "mount -t ramfs -o size=20M ramfs /tmp && iperf3 -p ${port} -s" > /dev/null && \
 	server_address=`$DOCKER_EXE inspect --format "{{.NetworkSettings.IPAddress}}" $($DOCKER_EXE ps -ql)` && \
 	$DOCKER_EXE run -ti --rm --name=iperf3-client ${image} bash -c "mount -t ramfs -o size=20M ramfs /tmp && iperf3 -c ${server_address} -d -t ${time}" > "$bidirectional_bandwidth_result"
-	total_bidirectional_bandwidth=`cat $bidirectional_bandwidth_result | tail -n 4 | head -1 | awk '{print $(NF-3), $(NF-2)}'`
+echo "Data:"
+cat $bidirectional_bandwidth_result
+	total_bidirectional_client_bandwidth=`cat $bidirectional_bandwidth_result | tail -n 3 | head -1 |  awk '{print $(NF-2), $(NF-1)}'`
```
Comment (jodh-intel) [inline tests/metrics/network/network-metrics-iperf3.sh]: Nit: double space before `awk`.

**Comment by jodh-intel** [comment]
Apart from a very trivial nit,

lgtm.

**Comment by chavafg** [comment]
qa-passed

**Comment by amshinde** [comment]
lgtm

File: tests/metrics/network/network-metrics-iperf3.sh [hunk 2]
```diff
@@ -58,24 +62,17 @@ function iperf3_jitter {
...
+# Run bi-directional TCP test, and extract results for both directions
+function iperf3_bidirectional_bandwidth_client_server {
 	setup
 	bidirectional_bandwidth_result=$(mktemp)
 	$DOCKER_EXE run -d --name=iperf3-server ${image} bash -c "mount -t ramfs -o size=20M ramfs /tmp && iperf3 -p ${port} -s" > /dev/null && \
 	server_address=`$DOCKER_EXE inspect --format "{{.NetworkSettings.IPAddress}}" $($DOCKER_EXE ps -ql)` && \
 	$DOCKER_EXE run -ti --rm --name=iperf3-client ${image} bash -c "mount -t ramfs -o size=20M ramfs /tmp && iperf3 -c ${server_address} -d -t ${time}" > "$bidirectional_bandwidth_result"
-	total_bidirectional_bandwidth=`cat $bidirectional_bandwidth_result | tail -n 4 | head -1 | awk '{print $(NF-3), $(NF-2)}'`
+echo "Data:"
+cat $bidirectional_bandwidth_result
```
Comment (jodh-intel) [inline tests/metrics/network/network-metrics-iperf3.sh]: These 2 lines need an indent.

General comment: these lines are very long - we should at some point try to fold them to fit within the constrains of a smaller terminal (also makes the code easier to maintain).

Comment (grahamwhaley) [inline tests/metrics/network/network-metrics-iperf3.sh]: In fact, those two lines that need an indent are debug accidentally left in - that'll teach me to push PRs at the end of the day without thoroughly reading the diff (rushing). Sigh, expect another version...
The long lines. Yes, they are. I'll not fold them in this PR, as it is just fixing up/shuffling things - but we should all take note for the future.

**Comment by grahamwhaley** [comment]
OK, let's try again...

**Comment by jodh-intel** [comment]
lgtm

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**